spec/graphql/relay/mutation_spec.rb

```diff
@@ -167,6 +167,70 @@
     end
   end
 
+  describe "specifying return interfaces" do
+    let(:result_interface) {
+      GraphQL::InterfaceType.define do
+        name "ResultInterface"
+        field :success, !types.Boolean
+        field :notice, types.String
+      end
+    }
+
+    let(:error_interface) {
+      GraphQL::InterfaceType.define do
+        name "ErrorInterface"
+        field :error, types.String
+      end
+    }
+
+    let(:mutation) {
+      interfaces = [result_interface, error_interface]
+      GraphQL::Relay::Mutation.define do
+        name "ReturnTypeWithInterfaceTest"
+
+        return_field :name, types.String
+
+        return_interfaces interfaces
+
+        resolve ->(obj, input, ctx) {
+          {
+            name: "Type Specific Field",
+            success: true,
+            notice: "Success Interface Field",
+            error: "Error Interface Field"
+          }
+        }
+      end
+    }
+
+    let(:schema) {
+      mutation_field = mutation.field
+
+      mutation_root = GraphQL::ObjectType.define do
+        name "Mutation"
+        field :custom, mutation_field
+      end
+
+      GraphQL::Schema.define do
+        mutation(mutation_root)
+        resolve_type ->(obj, ctx) { "not really used" }
```

Comment:
this is required by the schema checks which requires a `resolve_type`  if an interface is defined, but AFAICT it should be needed only if the interface is specified explicitly as a field type.